Source/Moya.swift

```diff
@@ -65,12 +87,16 @@ public class MoyaProvider<Target: TargetType> {
     }
     
     /// Returns an Endpoint based on the token, method, and parameters by invoking the endpointsClosure.
-    public func endpoint(token: Target) -> Endpoint<Target> {
+    public func endpoint(token: TargetType) -> Endpoint<TargetType> {
         return endpointClosure(token)
     }
     
     /// Designated request-making method. Returns a Cancellable token to cancel the request later.
     public func request(target: Target, completion: Moya.Completion) -> Cancellable {
+        guard let target = target.resource else {
```

Comment:
OK, that makes sense, thanks. I guess I was thinking of something like this:

```swift
enum StructTargetType: TargetType {
  public var path: String {
    switch self {
      case .Struct(let t): return t.path
    }
  }
  // and so on
}
```

So that the struct-based approach totally piggy-backed on the existing enum structure, instead of having the existing enum structure conform to something new. 

That was totally just _my_ interpretation, it may not be possible. You've played with this more than I have :smile: